src/helpers/buffers/ordering_sender.rs

```diff
@@ -97,7 +100,7 @@ impl State {
         if self.written > 0 && (self.written + self.spare.get() >= self.buf.len() || self.closed) {
             let v = self.buf[..self.written].to_vec();
             self.written = 0;
-
+            self.idle = false;
```

Comment:
just noticed that it is not 100% accurate. if buffers are very large, `take` won't happen for quite some time while there will be many `write` calls indicating that buffer is not idle.

Having said that, have you explored the avenue of letting code outside observe the buffer state and decide whether it is idle or no? It feels a much more scalable approach that does not require "observable" state to be plugged in buffer's code. It can be done with Deref polymorphism (wrapper) or by exploring buffer's internal state and comparing it with the latest snapshot. For send buffer it could be as simple as counting messages written to it and keeping this number as the state.